src/encode/mod.rs

```diff
@@ -28,6 +28,11 @@ const NEWLINE: &'static str = "\r\n";
 #[cfg(not(windows))]
 const NEWLINE: &str = "\n";
 
+thread_local!(
```

Comment:
add 
`![cfg(feature = "pattern_encoder")]`

Reply:
Oh, right. It's feature dependent. Thanks!